### PR TITLE
test(deploy): add OCM deploy tool

### LINKED ISSUE #657: Add a new deploytool to test Submariner using OCM

### PR DESCRIPTION
**Pull request by SteveMattar**
Test Submariner using OCM

* Rename the deploytool from olm to bundle
* Create common deploy functions
* Bump olm default version to v0.18.3
* Install `yq` in `shipyard-dapper-base image`
* Add ocm deploy tool
* The new deploy tool is now part of the CI

Closes #657

### CODE REVIEW

**Comment by SteveMattar** [comment]
Repository secrets `QUAY_USERNAME` `QUAY_PASSWORD` are missing.

File: scripts/shared/deploy.sh [hunk 1]
```diff
@@ -48,10 +48,105 @@ readonly CE_IPSEC_NATTPORT=4500
...
+# 2nd argument is the target namespace
+function create_operator_group() {
+  local og=$1
+  local ns=$2
+
+  echo "[INFO](${cluster}) Create the Operator group ${og}..."
+  # Create the OperatorGroup
+  render_template "${RESOURCES_DIR}"/common/operatorGroup.yaml | kubectl apply -f -
+
+  # Debug
```
Comment (Jaanki) [inline scripts/shared/deploy.sh:106]: This should run only in case of failure?

File: scripts/shared/deploy.sh [hunk 2]
```diff
@@ -48,10 +48,105 @@ readonly CE_IPSEC_NATTPORT=4500
...
+
+  # Manual Approve
+  echo "[INFO](${cluster}) Approve the installPlan..."
+  kubectl wait --for condition=InstallPlanPending --timeout=5m -n "${ns}" subs/"${sub}" || (echo "[ERROR](${cluster}) InstallPlan not found."; exit 1)
+  installPlan=$(kubectl get subscriptions.operators.coreos.com "${sub}" -n "${ns}" -o jsonpath='{.status.installPlanRef.name}')
+  if [ -n "${installPlan}" ]; then
+    kubectl patch installplan -n "${ns}" "${installPlan}" -p '{"spec":{"approved":true}}' --type merge
+  fi
+
+  # Debug
```
Comment (Jaanki) [inline scripts/shared/deploy.sh:137]: Ditto

File: scripts/shared/deploy.sh [hunk 3]
```diff
@@ -48,10 +48,105 @@ readonly CE_IPSEC_NATTPORT=4500
...
+
+# Create a CatalogSource
+# 1st argument is the catalogsource name
+# 2nd argument is the namespace
+# 3rd argument is index image url
+function create_catalog_source() {
+  local cs=$1
+  local ns=$2
+  # shellcheck disable=SC2034
+  local iib=$3
```
Comment (Jaanki) [inline scripts/shared/deploy.sh:72]: could be iiurl? iib does not seem to fit.

File: scripts/shared/deploy.sh (hunk 1 above)
Comment (SteveMattar) [reply scripts/shared/deploy.sh:106]: Creating the OperatorGroup is necessary for the submariner-operator namespace

File: scripts/shared/deploy.sh (hunk 3 above)
Comment (SteveMattar) [reply scripts/shared/deploy.sh:72]: iib refers to image index build and is how people at red hat call it
I will add a comment for making it more clear
thanks for the note

File: scripts/shared/deploy.sh (hunk 2 above)
Comment (SteveMattar) [reply scripts/shared/deploy.sh:137]: this is for approving the installplan after creating the subscription object
I can set it to auto-approve but find the manual option is more safe.

**Comment by SteveMattar** [comment]
https://github.com/submariner-io/shipyard/pull/677

**Pull request closed without merging**